src/main/java/org/eclipse/tractusx/ssi/lib/validation/JsonLdValidatorImpl.java

```diff
@@ -82,28 +98,17 @@ private void validateJsonLd(JsonLdObject jsonLdObject) throws InvalidJsonLdExcep
     }
   }
 
-  private static void findUndefinedTerms(JsonArray jsonArray) {
-    for (JsonValue entry : jsonArray) {
-      if (entry instanceof JsonObject) {
-        findUndefinedTerms((JsonObject) entry);
-      }
-    }
-  }
-
-  private static void findUndefinedTerms(JsonObject jsonObject) {
-    for (Map.Entry<String, JsonValue> entry : jsonObject.entrySet()) {
-      if (entry.getKey().startsWith(UNDEFINED_TERM_URI)) {
-
-        throw new RuntimeException(
-            "Undefined JSON-LD term: " + entry.getKey().substring(UNDEFINED_TERM_URI.length()));
-      }
-
-      if (entry.getValue() instanceof JsonArray) {
-        findUndefinedTerms((JsonArray) entry.getValue());
-      }
-      if (entry.getValue() instanceof JsonObject) {
-        findUndefinedTerms((JsonObject) entry.getValue());
+  public void validate(Verifiable verifiable) throws InvalidJsonLdException {
+    if (verifiable instanceof VerifiableCredential) {
+      validateJsonLd(verifiable);
+    } else if (verifiable instanceof VerifiablePresentation) {
+      VerifiablePresentation verifiablePresentation = (VerifiablePresentation) verifiable;
+      for (VerifiableCredential verifiableCredential :
+          verifiablePresentation.getVerifiableCredentials()) {
+        validate(verifiableCredential);
       }
+    } else {
+      LOG.warning("Unsupported Verifiable type: " + verifiable.getClass().getName());
```

Comment:
This will cause any `verifiable` that is neither a `VerifiableCredential` nor a `VerifiablePresentation` to get accepted (= to not raise an exception). 

This is the exact reason why you _don't_ use exceptions for error handling. And FYI, this exact bug has plagued the SSI lib in the past.

Reply:
Thanks @paullatzelsperger, I will take care of this issue and modify the code based on Paul comment.

@borisrizov-zf, please assign it to me.